eclass/qt6-build.eclass

```diff
@@ -133,48 +131,48 @@ qt6-build_src_install() {
 # @DESCRIPTION:
 # <flag> is the name of a flag in IUSE.
 qt_feature() {
-	[[ $# -ge 1 ]] || die "${FUNCNAME}() requires at least one argument"
-	echo "-DQT_FEATURE_${2:-$1}=$(usex $1 ON OFF)"
+	[[ ${#} -ge 1 ]] || die "${FUNCNAME}() requires at least one argument"
+
+	echo "-DQT_FEATURE_${2:-${1}}=$(usex ${1} ON OFF)"
 }
 
 # @FUNCTION: qt6_symlink_binary_to_path
 # @USAGE: <target binary name> [suffix]
 # @DESCRIPTION:
-# Symlink a given binary from QT6_BINDIR to QT6_PREFIX/bin, with optional suffix
+# Symlink a given binary from QT6_BINDIR to QT6_PREFIX/bin, with
+# optional suffix.
 qt6_symlink_binary_to_path() {
-	[[ $# -ge 1 ]] || die "${FUNCNAME}() requires at least one argument"
+	[[ ${#} -ge 1 ]] || die "${FUNCNAME}() requires at least one argument"
 
 	dosym -r "${QT6_BINDIR}"/${1} /usr/bin/${1}${2}
 }
 
 ######  Internal functions  ######
 
-# @FUNCTION: qt6_prepare_env
+# @FUNCTION: _qt6_prepare_env
 # @INTERNAL
 # @DESCRIPTION:
 # Prepares the environment for building Qt.
-qt6_prepare_env() {
+_qt6_prepare_env() {
 	# setup installation directories
 	# note: keep paths in sync with qmake-utils.eclass
-	QT6_PREFIX=${EPREFIX}/usr
-	QT6_HEADERDIR=${QT6_PREFIX}/include/qt6
-	QT6_LIBDIR=${QT6_PREFIX}/$(get_libdir)
-	QT6_ARCHDATADIR=${QT6_PREFIX}/$(get_libdir)/qt6
-	QT6_BINDIR=${QT6_ARCHDATADIR}/bin
-	QT6_PLUGINDIR=${QT6_ARCHDATADIR}/plugins
-	QT6_LIBEXECDIR=${QT6_ARCHDATADIR}/libexec
-	QT6_IMPORTDIR=${QT6_ARCHDATADIR}/imports
-	QT6_QMLDIR=${QT6_ARCHDATADIR}/qml
-	QT6_DATADIR=${QT6_PREFIX}/share/qt6
-	QT6_DOCDIR=${QT6_PREFIX}/share/qt6-doc
-	QT6_TRANSLATIONDIR=${QT6_DATADIR}/translations
-	QT6_EXAMPLESDIR=${QT6_DATADIR}/examples
-	QT6_TESTSDIR=${QT6_DATADIR}/tests
-	QT6_SYSCONFDIR=${EPREFIX}/etc/xdg
-	readonly QT6_PREFIX QT6_HEADERDIR QT6_LIBDIR QT6_ARCHDATADIR \
-		QT6_BINDIR QT6_PLUGINDIR QT6_LIBEXECDIR QT6_IMPORTDIR \
-		QT6_QMLDIR QT6_DATADIR QT6_DOCDIR QT6_TRANSLATIONDIR \
-		QT6_EXAMPLESDIR QT6_TESTSDIR QT6_SYSCONFDIR
+	readonly QT6_PREFIX=${EPREFIX}/usr
+	readonly QT6_DATADIR=${QT6_PREFIX}/share/qt6
+	readonly QT6_LIBDIR=${QT6_PREFIX}/$(get_libdir)
+
+	readonly QT6_ARCHDATADIR=${QT6_LIBDIR}/qt6
+
+	readonly QT6_BINDIR=${QT6_ARCHDATADIR}/bin
+	readonly QT6_DOCDIR=${QT6_PREFIX}/share/qt6-doc
+	readonly QT6_EXAMPLESDIR=${QT6_DATADIR}/examples
+	readonly QT6_HEADERDIR=${QT6_PREFIX}/include/qt6
+	readonly QT6_IMPORTDIR=${QT6_ARCHDATADIR}/imports
+	readonly QT6_LIBEXECDIR=${QT6_ARCHDATADIR}/libexec
+	readonly QT6_PLUGINDIR=${QT6_ARCHDATADIR}/plugins
+	readonly QT6_QMLDIR=${QT6_ARCHDATADIR}/qml
+	readonly QT6_SYSCONFDIR=${EPREFIX}/etc/xdg
+	readonly QT6_TESTSDIR=${QT6_DATADIR}/tests
+	readonly QT6_TRANSLATIONDIR=${QT6_DATADIR}/translations
```

Comment:
IIRC I wanted to use `declare -g -r` here but couldn't because we still had to support an older version of bash that didn't understand `-g`. That's no longer a concern now, though I dunno if using `declare` provides any advantages over `readonly`.

Reply:
Nah, beside that I find readonly kind of ugly :) But declare would be longer, so...